lib/influxdb2/client/worker.rb

```diff
@@ -94,22 +96,27 @@ def _check_background_queue(size: false)
 
     def _write(data)
       data.each do |key, points|
-        _write_raw(key, points)
+        _write_raw(key, points, 1, @write_options.retry_interval)
       end
     end
 
-    def _write_raw(key, points)
+    def _write_raw(key, points, attempts, retry_interval)
       if @write_options.jitter_interval > 0
         jitter_delay = (@write_options.jitter_interval.to_f / 1_000) * rand
         sleep jitter_delay
       end
       @api_client.write_raw(points.join("\n"), precision: key.precision, bucket: key.bucket, org: key.org)
     rescue InfluxError => e
-      raise e if e.code.nil? || !(%w[429 503].include? e.code)
+      raise e if e.code.nil? || !(%w[429 503].include? e.code) || attempts > @write_options.max_retries
+
+      timeout = if e.retry_after.empty?
```

Comment:
The `jitter_interval` should be append to timeout if it's specified. Please also add tests for that.